### ISSUE TITLE
anko-sqlite:castValue for Boolean type  always return true

### ISSUE THREAD

**Issue by mitty1987** [Kotlin/anko#441]
if (value is Long) {
        if (type == java.lang.Integer.TYPE || type == java.lang.Integer::class.java) {
            return value.toInt()
        } else if (type == java.lang.Short.TYPE || type == java.lang.Short::class.java) {
            return value.toShort()
        } else if (type == java.lang.Byte.TYPE || type == java.lang.Byte::class.java) {
            return value.toByte()
        } else if (type == java.lang.Boolean.TYPE || type == java.lang.Boolean::class.java) {
            return value != 0// here  0!=0 is true
        } else if (type == java.lang.Character.TYPE || type == java.lang.Character::class.java) {
            return value.toChar()
        }
    }

**Comment by zarmhast** [comment]
```Kotlin
data class Class13(val id: Int, val b1: Boolean, val b2: Boolean)
classParser<Class13>().parseRow(arrayOf(123L, 0L, 1L))
// Class13(id=123, b1=true, b2=true
```

It happens because the test is Long(0) != 0. Which due to some Java boxing rules ends up being true because Long != Int.

Solutions:

```Kotlin
return value.toInt() != 0
return value != 0L
```

**Comment by 4u7** [comment]
Fixed with bebbc8ae71063502005a95c4a306cf792998bde4
